gigascience/gigadb-website
# A11y - Accessible header search bar

# Pull request for issue: #1364

This is a pull request for the following functionalities:

![image](https://github.com/gigascience/gigadb-website/assets/143437854/c97af630-cbdd-4a0f-b1c7-2b2f62690d9f)


* Reorder search bar and social links so that visual order and tab order are the same
* Enhance focus states of search input and search button
* Improve accessibility of search form by labelling the search form and the input
* Apply changes to each layout with a top search bar

## How to test?

* Navigate with keyboard and check that elements get visible focus in visual order in the following pages:
  * http://gigadb.gigasciencejournal.com:9170/
  * http://gigadb.gigasciencejournal.com:9170/dataset/100006
  * http://gigadb.gigasciencejournal.com:9170/site/faq

## How have functionalities been implemented?

* The search input was repositioned in the dom before the social links. To restore the layout, both elements are wrapped in a flex box. For small screens, the search input is repositioned underneath the social icons (I guess another alternative would be to hide it since there's already a search input in the hero banner

## Any issues with implementation?

* I'm not sure where the uploader_layout is used but it seemed to use the same top bar + search input, so it felt safe to update it. If someone points me to the page where this layout is used, I will cross check it

NOTE: The issue mentioned below is temporarily solved by changing the order in which acceptance tests run (by renaming files)

* Running `./tests/acceptance_runner` some tests related to the search fail. When I run the failing tests in isolation, they pass. When I run the failing test first (by renaming it to `000-`), it passes

These are the failing tests:

```sh
E main search function: basic search (0.27s)
E main search function: pagination show correct number of pages (0.22s)
E main search function: Can navigate to the next page (0.29s)
E main search function: Can search compound term without operator (0.28s)
E main search function: Can search compound term with operator (0.27s)
E main search function: Can search compound term with double quotes (0.30s)
E main search function: can limit search to specific year (0.30s)
E main search function: can limit search to specific month (0.27s)
E main search function: can limit search to specific day (0.29s)
E main search function: Limit results to datasets (0.25s)
E main search function: Limit results to samples (0.26s)
E main search function: Limit results to files (0.24s)
E main search function: Show a message when nothing is found (0.28s)
E main search function: Search for a term that is only in dataset types (0.31s)
E : Hide the Save current search criteria option (0.72s)
```

If I run this `docker-compose run --rm codecept run --no-redirect -g ok-can-offline acceptance tests/acceptance/Search.feature:basic\ search`, the tests pass

That suggests to me that the tests might be interdependent somehow

Is anyone, on first glance, able to give me some pointers? Did anyone encounter this problem before?

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
